File: Laba 4 c sharp/Ryadok.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ClassRyadok
{
	public class Ryadok
	{
		private char[] _ryadok = { };//масив чарів - string 
		private int _len;//довжина рядка
		public Ryadok()
		{
			this._ryadok = new char[0];
			_len = 0;
		}

		public Ryadok(char[] ryadok)//Конструктор з параметрами
		{
			_len = Dovgina(ryadok);
			_ryadok = ryadok;
		}

		public Ryadok(Ryadok other)//Конструктор копіювання
		{
			_len = other._len;
			_ryadok = other._ryadok;

		}
		public int Dovgina(char[] str)//знаходження довжини рядка
		{
			int dovgina = 0;
			while (str[dovgina] != '\0')
			{
				dovgina++;
			}
			return dovgina;
		}
		public static Ryadok operator +(Ryadok left, Ryadok other)//перенавантаження оператора додавання
		{
			Ryadok suma = new Ryadok();
			suma._ryadok = new char[left._len + other._len + 1];
			for (int i = 0; i < left._len; i++)
			{
				suma._ryadok[i] = left._ryadok[i];
			}
			for (int j = left._len; j < left._len + other._len + 1; j++)
			{
				suma._ryadok[j] = other._ryadok[j - left._len];
			}

			return suma;
		}
		public static Ryadok operator -(Ryadok left, char vidyemnuk)
		{
			int numOfnull = 0;
			for (int i = 0; i < left._len; i++)
			{
				if (left._ryadok[i] == vidyemnuk)
				{
					numOfnull += 1;
				}

			}
			Ryadok newLeft = new Ryadok(new char[left._len - numOfnull]);
			int j = 0;
			for (int i = 0; i < left._len; i++)
			{
				newLeft._ryadok[j] = left._ryadok[i];
				if (left._ryadok[i] == vidyemnuk)
				{
					continue;
				}
				j++;
			}
			left._len -= numOfnull;
			return newLeft;
		}
		public void Print()
		{
			Console.WriteLine(this._ryadok);
		}

	}
}

File: Laba_4_c++/Ryadok.cpp
#include<iostream>
#include"Ryadok.h"



Ryadok::Ryadok()//Конструктор без параметрів(для створення пустого обєкту)
	{
		_ryadok = nullptr;
		_len = 0;
	}
Ryadok::Ryadok(const char* ryadok)//Конструктор з параметрами
	{
		_len = Dovgina(ryadok);
		_ryadok = new char[_len + 1];
		for (int i = 0; i < _len; i++)
		{
			_ryadok[i] = ryadok[i];
		}
		_ryadok[_len] = '\0';
	}
Ryadok::Ryadok(const Ryadok& other)//Конструктор копіювання
	{
		_len = other._len;
		_ryadok = new char[other._len + 1];
		for (int i = 0; i < other._len + 1; i++)
		{
			_ryadok[i] = other._ryadok[i];
		}
			   
}
int Ryadok::Dovgina(const char* str)//знаходження довжини рядка
	{
		int dovgina = 0;
		while (str[dovgina] != '\0')
		{
			dovgina++;
		}
		return dovgina;
	}
Ryadok Ryadok:: operator +(const Ryadok& other)//перенавантаження оператора додавання
	{
		char* ryadok_sum = new char[_len + other._len + 1];
		for (int i = 0; i < _len; i++)
		{
			ryadok_sum[i] = _ryadok[i];
		}
		for (int j = _len; j < _len + other._len + 1; j++)
		{
			ryadok_sum[j] = other._ryadok[j - _len];
		}

		return ryadok_sum;
	}
Ryadok& Ryadok:: operator=(const Ryadok& other)
	{
		_len = other._len;
		delete[]_ryadok;
		_ryadok = new char[other._len + 1];
		for (int i = 0; i < other._len; i++)
		{
			_ryadok[i] = other._ryadok[i];
		}
		_ryadok[other._len] = '\0';
		return *this;
	}

Ryadok& Ryadok:: operator-(const char vidyemnuk)
	{
		int num_of_null = 0;
		for (int i = 0; i < _len; i++)
		{
			if (_ryadok[i] == vidyemnuk)
			{
				num_of_null += 1;
			}
		}
		char* new_ryadok = new char[_len - num_of_null + 1];
		int j = 0;
		for (int i = 0; i < _len; i++)
		{
			new_ryadok[j] = _ryadok[i];
			if (_ryadok[i] == vidyemnuk)
			{
				continue;
			}
			j++;
		}
		new_ryadok[_len - num_of_null] = '\0';
		_len -= num_of_null;
		_ryadok = new_ryadok;
		return *this;
	}
void Ryadok:: Print()
	{
		std::cout << _ryadok << std::endl;
	}

Ryadok::~Ryadok()//Диструктор
	{
		delete[]_ryadok;
	}
